### DISCUSSION TITLE
Transfer ClaimData Between Servers?

### DISCUSSION THREAD

**Discussion by CableCabe** [GriefPrevention/GriefPrevention#797] [Q&A] [answered]
Hi, is there a way to transfer claimdata between two servers? When I drag the files into the ClaimData folder in the new server, it just deletes them. Thanks!

**Comment by Jikoo** [comment]
Is the second server a fresh GP install or does it already have claims? Is the server shut down while you try this? Did you remember to transfer the file containing the last claim ID? Is the world name the same? Did you enable claims for the world?

There's no officially supported way to transfer data like that, but as long as claim IDs don't conflict, the world name is the same, and the last claim ID is higher than any existing claim IDs it should be fine.

**Comment by CableCabe** [comment]
Not a new GP install, we just transfered the whole GP folder from the last server over. We've restarted the server several times but no dice. I've tried completely replace all of the ClaimData and PlayerData files and they just keep getting deleted everytime we reload/restart.

What exactly do you mean about the last claim ID? I'm not very well versed with this plugin. 

The world name was not the same, but I changed the name in the main config file to the current world. It's the same world file/data just a different folder name (had to change due to our host).

**Comment by Jikoo** [comment, marked as answer]
Ah, that's a problem then. Okay, so, if you're using a database I can't help you, but if you're using flatfile (based on your response, you are) we can wrap your issue with a bit of scripting.

GP stores its claims in yaml files by claim ID - 1.yml, 2.yml, and so on. Inside the yaml file is all of the data of the claim - boundaries, trust, etc.
I.E. here's 3.yml from my own server:
```yaml
Lesser Boundary Corner: world;-106;58;-185
Greater Boundary Corner: world;-86;59;-165
Owner: ''
Builders: []
Containers: []
Accessors: []
Managers: []
Parent Claim ID: -1
inheritNothing: false
```
_nextClaimID is also a file contained in the claim data folder, contains the next unused claim ID. Should be 1 higher than your highest claim ID, but that's unlikely to be a concern.
The real issue is that, as you can see, the claims know what world they're in. Since you changed world names, GP is a) no longer enabled for the world and b) is trying to apply the claims to the wrong world.

For a quick fix, you can simply run the following bash script (with the correct info edited in):
```bash
#!/bin/bash

if [ $# -lt 2 ]; then
    echo 'Usage: <old world name> <new world name>'
    exit 0
fi

path="/path/to/server/plugins/GriefPreventionData/ClaimData/"

original=$1
new=$2

cd $path

find . -type f -name "*.yml" | while read ymlfile ; do
    sed -i "s/${original}/${new}/g" $ymlfile
done
```
You could probably compact that into a 1-liner pretty easily, but frankly I'm no great shakes at bash and this was just an adaptation of a script I wrote for migrating a player's UUID after someone swapped accounts.

**Comment by CableCabe** [comment]
Thank you very much! Changing the world name in the first two lines of the claimdata fixed my issue!
